src/telliot/datasource.py

```diff
@@ -45,13 +47,13 @@ def latest(self) -> OptionalDataPoint[T]:
         else:
             return None, None
 
-    def store_datapoint(self, datapoint: DataPoint[T]) -> None:
+    def store_datapoint(self, datapoint: OptionalDataPoint[T]) -> None:
         """Store a datapoint"""
         v, t = datapoint
         if v is not None and t is not None:
             self._history.append(datapoint)
 
-    def get_all_datapoints(self) -> List[DataPoint[T]]:
+    def get_all_datapoints(self) -> List[OptionalDataPoint[T]]:
```

Comment:
This should stay as DataPoint, because there should only be valid datapoints in the history.

Reply:
Reverted changes in `datasource.py`. Made a change to storing in the ampl source file instead